src/screens/ImportSeedPhraseSheet.js

```diff
@@ -247,18 +254,38 @@ export default function ImportSeedPhraseSheet() {
     if (!wasImporting && isImporting) {
       startAnalyticsTimeout(async () => {
         const input = resolvedAddress ? resolvedAddress : seedPhrase.trim();
+        const previousWalletCount = !isObjectEmpty(wallets)
+          ? Object.keys(wallets).filter(
+              wallet => wallet.type !== WalletTypes.readOnly
+            ).length
+          : 0;
         initializeWallet(input, color, name ? name : '')
           .then(success => {
             handleSetImporting(false);
             if (success) {
-              setTimeout(() => {
-                navigate(Routes.WALLET_SCREEN);
-              }, 300);
-              if (Platform.OS === 'android') {
-                hide();
-              }
-              analytics.track('Imported seed phrase', {
-                hadPreviousAddressWithValue,
+              goBack();
+              InteractionManager.runAfterInteractions(async () => {
+                if (previousWalletCount === 0) {
+                  await saveUserBackupState(BackupStateTypes.done);
+                  replace(Routes.SWIPE_LAYOUT);
+                } else {
+                  navigate(Routes.WALLET_SCREEN);
+                }
+                if (Platform.OS === 'android') {
+                  hide();
+                }
+
+                setTimeout(() => {
+                  // If it's not read only, show the backup sheet
+                  if (!(isENSAddressFormat(input) || isValidAddress(input))) {
+                    Navigation.handleAction(Routes.BACKUP_SHEET, {
+                      option: 'imported',
+                    });
+                  }
+                }, 1000);
+                analytics.track('Imported seed phrase', {
+                  hadPreviousAddressWithValue,
```

Comment:
hadPeviousAddressWithValue may be a misleading name since it represents whether or not the wallet had zero ETH (as in, if it had zero ETH, then useIsWalletEthZero will return true)